pkg/provisioner/nodeprovisioner.go

```diff
@@ -44,13 +48,40 @@ type ioHandler interface {
 	ReadFile(filename string) ([]byte, error)
 }
 
+type deviceChecker struct {
+	diskURI         string
+	curRetry        int
+	maxRetry        int
+	lock            sync.RWMutex
+	detachInProcess bool
+}
+
+func (d *deviceChecker) backoff(diskURI string) {
+	d.lock.Lock()
+	defer d.lock.Unlock()
+	if diskURI == d.diskURI {
+		d.maxRetry *= defaultBackoffScale
```

Comment:
Uh, `defaultMaxRetry` is now `0`, so what is the purpose of this function? Do we even need to backoff at this point?

Reply:
Oh wait, I meant to set the value to `1` and start with `curRetry = 1` so that recovery can kick in on the first failure.

I thought we wanted to keep the back off in case there is an outage causing this to fall into an indefinite loop?

Reply:
I'm not sure that backing off due to outage is helpful at this point. If the initial attempt to start recovery is unsuccessful, I do not believe it necessary to backoff. The outage will fail future recovery operations until it is resolve. K8s (I assume) will give up at some point. It doesn't really matter whether we fail due to backoff or due to an ongoing outage. I don't see that backoff really helps here.